docs/pages/contributing.md

```diff
@@ -0,0 +1,99 @@
+# Contributing
+
+## Developer Guide
+
+### Setup
+
+This project uses [hatch](https://hatch.pypa.io/latest/) to manage the development environment and build and publish releases. Make sure `hatch` is [installed](https://hatch.pypa.io/latest/install/) first:
+
+```bash
+$ pip install hatch
+```
+
+Now you can [enter the development environment](https://hatch.pypa.io/latest/environment/#entering-environments) using:
+
+```bash
+$ hatch shell
+```
+
+This will install development dependencies in an isolated environment and drop you into a shell (use `exit` to leave).
+
+### Pre-commit
+
+Use [pre-commit](https://pre-commit.com/) to run linting, type-checking, and formatting:
+
+```bash
+$ pre-commit run --all-files
+```
+
+...or install it to run automatically before every commit with:
+
+```bash
+$ pre-commit install
+```
+
+You can run pre-commit hooks separately and pass additional arguments to them. For example, to run `black` on a single file:
+
+```bash
+$ pre-commit run black --files=src/sknnr/_base.py
+```
+
+### Testing
+
+Unit tests are *not* run by `pre-commit`, but can be run manually using `hatch` [scripts](https://hatch.pypa.io/latest/config/environment/overview/#scripts):
+
+```bash
+$ hatch run test:all
+```
+
+Measure test coverage with:
+
+```bash
+$ hatch run test:coverage
+```
+
+Any additional arguments are passed to `pytest`. For example, to run a subset of tests matching a keyword:
+
+```bash
+$ hatch run test:all -k gnn
+```
+
+### Documentation
+
+Documentation is built with [mkdocs](https://www.mkdocs.org/). During development, you can run a live-reloading server with:
+
+```bash
+$ hatch run docs:serve
+```
+
+To build the documentation locally, use:
+
+```bash
+$ hatch run docs:build
+```
+
+!!! warning
+
+    The built docs do not need to be committed, as they are automatically built upon deployment to ReadTheDocs.
```

Comment:
I'm *pretty sure* this is true, but will need to confirm once we have RTD working.